### PR TITLE
feat(pandas): Flopy pandas support

### PR DESCRIPTION
**Pull request by spaulins-usgs**
This PR is the first step towards integrating Pandas into Flopy.  This integration takes place in the MFPandasList and MFPandasTransientList classes (MFPandas*), which are used instead of the MFList and MFTransientList classes under the following conditions:

1)	The data is in a package is not an considered advanced
2)	The data type meets certain criteria, like it is not a jagged list with variable column length
3)	The flopy simulation data option use_pandas is set to true (default value is true)

The MFPandas* classes currently support the same interface as MFList and MFTransientList, and should behave similarly to the end-user.  However, MFPandas* stores data internally in a Pandas Dataframe and reads and writes data using Pandas “read_csv” and “to_csv” methods, which can be significantly faster than flopy’s current file reading.  The MFPandas* classes set_data methods support DataFrames and their new “get_dataframe” method returns data in a Panda’s Dataframe (“get_data” still returns a recarray).

Remaining work on this PR includes:

1)	When reading files do not use python’s “tell” method to record the start and finish of data (this can be problematic for files opened as text).
2)	Convert recarrays to Pandas using the “from_records” method instead of the dataframe constructor.
3)	Remove cellid tuples support from Flopy.  Flopy will only accept cellids stored in separate layer, row, and column fields (or appropriate fields for the discretization) instead of also supporting cellids as a single field with a tuple (layer, row, column).  All flopy lists (including the old MFList* classes and the new MFPandasList* classes) will store each component of the cellid in a separate column.  This feature may or may not be part of this PR depending on timing.

### CODE REVIEW

File: flopy/mf6/data/dfn/gwf-buy.dfn [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # --------------------- gwf buy options ---------------------
+# package-type stress-package
```
Comment (langevin-usgs) [inline flopy/mf6/data/dfn/gwf-buy.dfn:2]: The Buoyancy Package is not a stress package, so perhaps this terminology is not correct here.

File: flopy/mf6/data/dfn/gwf-csub.dfn [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # --------------------- gwf csub options ---------------------
+# package-type advanced-stress-package
```
Comment (langevin-usgs) [inline flopy/mf6/data/dfn/gwf-csub.dfn:2]: CSUB is not a stress package.

**Review by langevin-usgs — COMMENTED** [review]
Couple quick notes about buoyancy and csub -- these are not really stress packages, so I'm wondering if the terminology needs to be modified.

**Comment by wpbonelli** [comment]
A few quick thoughts:

- If flopy is to accept tabular data as a) lists of lists/tuples, b) numpy recarrays, or c) pandas dataframes, couldn't it accept any of these without needing `use_pandas=True` at `MFSimulation` init/load. Is there any reason the user might want to disable pandas support now that pandas is a core dependency? If the motivation to expose this option is performance, are there any reasons not to use pandas if it's generally faster? I.e. case-specific tradeoffs?

- It looks like `get_dataframe()` is not supported on `MFList`. Also, while `MFPandasList.get_data()` returns a recarray, it looks like `MFList.get_data()` may return a recarray or a list (is this correct?). Seems worthwhile to keep user-facing APIs consistent regardless of flopy's internal data model.

- Points above seem to raise the question, are separate MFPandas* classes needed? If Pandas consistently performs better, and both approaches are to be functionally equivalent, could we fold pandas support into the existing classes?

**Comment by wpbonelli** [comment]
I closed this by mistake. I meant to push to `scottrp/develop` (related to the [model removal fix](https://github.com/modflowpy/flopy/pull/1945) merged yesterday), but I mixed up my remotes and pushed to `spaulins-usgs/develop` instead.

No work was lost &mdash; both @spaulins-usgs and I had local backups. If GitHub does not allow reopening this PR, and a replacement PR is necessary, maybe worth moving the comments over, or linking back here. Hopefully some backend service will refresh, notice `spaulins-usgs/develop` *does* have the [new changeset](https://github.com/spaulins-usgs/flopy) again, and allow reopening.

Lesson learned: don't push to someone else's PR branch, even if GitHub allows it. I am sorry all.

File: flopy/mf6/data/dfn/gwf-buy.dfn (hunk 1 above)
Comment (spaulins-usgs) [reply flopy/mf6/data/dfn/gwf-buy.dfn]: removed "stress-package" from buoyancy package dfn

File: flopy/mf6/data/dfn/gwf-csub.dfn (hunk 2 above)
Comment (spaulins-usgs) [reply flopy/mf6/data/dfn/gwf-csub.dfn]: removed "stress-package" from CSUB package dfn

**Pull request closed without merging**